### PR TITLE
Revamp blog page look & feel.

### PR DESCRIPTION
**Pull request by geeknoid**
- The blog page now has a sidebar showing all blog posts.

- Blog entries now have a clean view of publication date, attribution, and subtitle

- Sidebar entries, both on the blog page and in the docs area now show tooltips when hovering the
mouse over the entry.

Staging site: https://geeknoid.github.com/istio.github.io

PS: Please give the staging site a good run-through so make sure I didn't mess up any edge cases.

### CODE REVIEW

File: _data/istio.yml [hunk 1]
```diff
@@ -1 +1,2 @@
 version: 0.3.0
+latest_blog_post: /blog/posts/2017/mixer-spof-myth.html
```
Comment (ldemailly) [inline _data/istio.yml]: add the missing newline?

File: _docs/concepts/index.md [hunk 2]
```diff
@@ -10,4 +10,5 @@ type: markdown
 
 ---
 
+{% assign docs = site.docs %}
```
Comment (ldemailly) [inline _docs/concepts/index.md]: what does this do / why is it needed ? what happens if you missed one ?

**Comment by rshriram** [comment]
missing image in https://geeknoid.github.io/istio.github.io/blog/posts/2017/0.1-using-network-policy.html

File: _blog/posts/2017/0.1-auth.md [hunk 3]
```diff
@@ -1,12 +1,17 @@
 ---
-title: "Istio Auth: secure by default service to service communications"
+title: "Istio Auth"
```
Comment (rshriram) [inline _blog/posts/2017/0.1-auth.md]: rename to Using Istio to improve end-to-end Security ?

File: _blog/posts/2017/0.1-announcement.md [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
-title: "Introducing Istio: A robust service mesh for microservices"
+title: "Introducing Istio"
 overview: Istio 0.1 announcement
-published: true
-permalink: blog/istio-service-mesh-for-microservices.html
-attribution: The Istio Team
-layout: post
+publish_date: May 24, 2017
+subtitle: A robust service mesh for microservices
+attribution:
```
Comment (rshriram) [inline _blog/posts/2017/0.1-announcement.md]: Martin Taill.., Louis Ryan, Shriram Rajagopalan. ??

File: _blog/posts/2017/0.1-canary.md [hunk 5]
```diff
@@ -1,11 +1,15 @@
 ---
 title: "Canary Deployments using Istio"
 overview: Using Istio to create autoscaled canary deployments
-published: true
-permalink: blog/canary-deployments-using-istio.html
-attribution: The Istio Team
-layout: post
+publish_date: June 14, 2017
+subtitle:
+attribution:
```
Comment (rshriram) [inline _blog/posts/2017/0.1-canary.md]: Frank Budinsky

Comment (rshriram) [reply _blog/posts/2017/0.1-canary.md]: Actually, Frank Budinsky, IBM

File: _blog/posts/2017/0.1-announcement.md (hunk 4 above)
Comment (rshriram) [reply _blog/posts/2017/0.1-announcement.md]: Possibly X, Company format? Or should we dump the company?

File: _blog/posts/2017/0.1-using-network-policy.md [hunk 6]
```diff
@@ -1,11 +1,15 @@
...
-title: "Using Network Policy in concert with Istio"
+title: "Using Network Policy with Istio"
 overview: How Kubernetes Network Policy relates to Istio policy
-published: true
-permalink: blog/using-network-policy-in-concert-with-istio.html
-attribution: The Istio Team
-layout: post
+publish_date: August 10, 2017
+subtitle:
+attribution:
```
Comment (rshriram) [inline _blog/posts/2017/0.1-using-network-policy.md]: Spike Curtis

**Comment by ldemailly** [comment]
if you add the missing trailing / to https://www.weave.works/blog/istio-weave-cloud/ it should resolve the circleci failure

but I wonder how come it doesn't flag the missing image / is it some JS instead of plain html ?

File: _blog/posts/2017/0.1-announcement.md (hunk 4 above)
Comment (ldemailly) [reply _blog/posts/2017/0.1-announcement.md]: I think "The Istio Team" is more appropriate than 3 individuals for the 0.1 announcement

File: _data/istio.yml (hunk 1 above)
Comment (geeknoid) [reply _data/istio.yml]: Whoops, that was a failed experiment, so I just reverted the change completely.

File: _docs/concepts/index.md (hunk 2 above)
Comment (geeknoid) [reply _docs/concepts/index.md]: Turns the sidebar functionality into a function. Whereas before, the sidebar was closely coupled with our "docs" tree, it's now more general-purpose and can be used with any hierarchy of documents (such as blogs).

If you omit this statement, you'll end up with a blank sidebar.

File: _blog/posts/2017/0.1-announcement.md (hunk 4 above)
Comment (geeknoid) [reply _blog/posts/2017/0.1-announcement.md]: I went with "The Istio Team" for the release announcements.

File: _blog/posts/2017/0.1-using-network-policy.md (hunk 6 above)
Comment (geeknoid) [reply _blog/posts/2017/0.1-using-network-policy.md]: Done

**Comment by geeknoid** [comment]
@ldemailly The link tracking appears to have some issues. You mentioned here to add a trailing / to fix the weavecloud link. I added the link as requested and the link works fine. The issue is that I hadn't touched that text at all, I just moved the blog post to a different location.

And as you point out, it didn't flag the missing image. It's conceivable that the link checker is only looking at <a href> and not .

**Pull request merged**